service/src/main/java/bio/terra/workspace/app/controller/FolderApiController.java

```diff
@@ -107,8 +107,9 @@ public ResponseEntity<ApiFolderList> listFolders(UUID workspaceId) {
   @Override
   public ResponseEntity<Void> deleteFolder(UUID workspaceId, UUID folderId) {
     AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
-    workspaceService.validateWorkspaceAndAction(userRequest, workspaceId, SamWorkspaceAction.WRITE);
-    folderService.deleteFolder(workspaceId, folderId);
+    // Only
+    workspaceService.validateWorkspaceAndAction(userRequest, workspaceId, SamWorkspaceAction.OWN);
```

Comment:
Is this restriction OK with Product/UX? Are WRITERS allowed to create folders? Should it be symmetric?

It would be possible to allow writers to delete, but it would require a recursive check of all resources within the folder to make sure that the operation was possible. An alternate semantic would be that the delete would succeed in deleting everything the user had access to.

nit: incomplete comment above

Reply:
Here's what I think should happen (can be separate PR):

- Writer can delete folder
- If there's a private resource (not owned by requester), then deletion fails with something like `Cannot delete folder because XX does not have permission to delete private resource YY. Ask workspace owner or ZZ to delete YY first.`
- Otherwise, deletion succeeds.